vfs/vfs_test.go

```diff
@@ -342,12 +342,7 @@ func TestVFSRootDirName(t *testing.T) {
 		require.NoError(t, err)
 		fi, err := rootDir.Stat()
 		require.NoError(t, err)
-
-		exp := sep
-		if fs == Default {
-			exp = string(os.PathSeparator)
-		}
-		require.Equal(t, exp, fi.Name())
+		require.Equal(t, sep, fi.Name())
```

Comment:
This was a bit of head scratcher. This test was failing on Windows only, returning a `/` instead of the expected `\` (Windows path sep).

Digging into the Go runtime changes in 1.20, the behavior of `os.Stat` changed in https://github.com/golang/go/commit/0f0aa5d8a6a0253627d58b3aa083b24a1091933f. I don't know enough about the Windows API to state with confidence why that commit results in returning `/`, but it seems like the previous behavior wasn't using the file handle to imply the file name, which was problematic (possible TOCTOU bug).

Funnily enough, spelunking in our codebase revealed that @ajwerner was running up against a "rather thorny" issue on the original PR (see https://github.com/cockroachdb/pebble/pull/1435#pullrequestreview-843827694), which resulted in the need for the special casing here.

This simplified version seems to do the trick and satisfy the Windows tests.